fedimint-bin-tests/src/util.rs

```diff
@@ -14,17 +14,42 @@ use super::*;
 #[derive(Debug, Clone)]
 pub struct ProcessHandle(Arc<ProcessHandleInner>);
 
+impl ProcessHandle {
+    pub async fn kill_process(self) -> Result<()> {
+        let ProcessHandle(arc_process_handle_inner) = self;
+        let process_handle_inner = &mut Arc::try_unwrap(arc_process_handle_inner).unwrap();
+        let mut child = std::mem::take(&mut process_handle_inner.child)
+            .ok_or_else(|| anyhow!("Unable to own ProcessHandleInner"))?;
```

Comment:
please rewrite this, i don't see a reason to mem::take here.

Reply:
Oh I missed that. I was dropping `ProcessHandle` and therefore I used `take`, or else it would have dropped, and we would not be able to use the Child to wait for shutdown.

Reply:
I have explained it in a comment bellow